src/networkcreator.js

```diff
@@ -554,6 +554,8 @@ class NetworkCreator extends EventEmitter {
     } else {
       this.hasGeth = true
     }
+    output.gethFaucet = this.generateGethFaucet(volumes)
```

Comment:
Geth related services should only be added in on chain mode, so maybe move this to the if `this.hasGeth` statement above it.